### PR TITLE
[DSIP-42] Add dolphinscheduler-aws-authentication module 

### LINKED ISSUE #16024: [DSIP-42] Add dolphinscheduler-aws-authentication module
### Search before asking

- [X] I had searched in the [DSIP](https://github.com/apache/dolphinscheduler/issues/14102) and found no similar DSIP.

### Motivation

Right now, we have multiple module might need to login with aws e.g. s3-storage, EMR task....
It's need to add an `aws-authentication` module to provide a service help to do the login logic.

### Design Detail

- Add new module dolphinscheduler-authentication, which is a pom module.
- Add new module dolphinscheduler-aws-authentication under dolphinscheduler-authentication
- Move aws authentication code into dolphinscheduler-aws-authentication
- Move aws config into aws.yaml

### Compatibility, Deprecation, and Migration Plan

Compatibility with latest version

### Test Plan

Test by manual

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by ruanwenjun**
## Purpose of the pull request

close https://github.com/apache/dolphinscheduler/issues/16024

## Verify this pull request

This pull request is code cleanup without any test coverage.

*(or)*

This pull request is already covered by existing tests, such as *(please describe tests)*.

(or)

This change added tests and can be verified as follows:

(or)

If your pull request contain incompatible change, you should also add it to `docs/docs/en/guide/upgrede/incompatible.md`

### CODE REVIEW

File: deploy/kubernetes/dolphinscheduler/values.yaml [hunk 1]
```diff
@@ -250,20 +250,25 @@ conf:
     # -- resource store on HDFS/S3 path, resource file will store to this base path, self configuration, please make sure the directory exists on hdfs and have read write permissions. "/dolphinscheduler" is recommended
     resource.storage.upload.base.path: /dolphinscheduler
 
-    # -- The AWS access key. if resource.storage.type=S3 or use EMR-Task, This configuration is required
-    resource.aws.access.key.id: minioadmin
+    # The AWS credentials provider type. support: AWSStaticCredentialsProvider, InstanceProfileCredentialsProvider
+    # AWSStaticCredentialsProvider: use the access key and secret key to authenticate
+    # InstanceProfileCredentialsProvider: use the IAM role to authenticate
+    credentials.provider.type: AWSStaticCredentialsProvider
```
Comment (Gallardot) [inline deploy/kubernetes/dolphinscheduler/values.yaml]:
```suggestion
    aws.credentials.provider.type: AWSStaticCredentialsProvider
```

Comment (Gallardot) [reply deploy/kubernetes/dolphinscheduler/values.yaml]: Since this configuration is used by aws, it is recommended to add the aws prefix

Comment (Gallardot) [reply deploy/kubernetes/dolphinscheduler/values.yaml]: opps, looks like a typo

File: docs/docs/en/guide/resource/configuration.md [hunk 2]
```diff
@@ -28,74 +28,37 @@ The configuration you may need to change:
 
 ## connect AWS S3
 
-if you want to upload resources to `Resource Center` connected to `S3`, you need to configure `api-server/conf/common.properties` and `worker-server/conf/common.properties`. You can refer to the following:
+if you want to upload resources to `Resource Center` connected to `S3`, you need to configure `api-server/conf/common.properties`, `api-server/conf/s3.yaml` and `worker-server/conf/common.properties`, `worker-server/conf/s3.yaml`. You can refer to the following:
```
Comment (Radeity) [inline docs/docs/en/guide/resource/configuration.md]: s3.yaml -> aws.yaml

File: docs/docs/zh/guide/resource/configuration.md [hunk 3]
```diff
@@ -26,77 +26,35 @@ Dolphinscheduler 资源中心使用本地系统默认是开启的，不需要用
 
 ## 对接AWS S3
 
-如果需要使用到资源中心的 S3 上传资源，我们需要对以下路径的进行配置：`api-server/conf/common.properties` 和 `worker-server/conf/common.properties`。可参考如下：
+如果需要使用到资源中心的 S3 上传资源，我们需要对以下路径的进行配置：`api-server/conf/common.properties`, `api-server/conf/s3.yaml` 和 `worker-server/conf/common.properties`, `worker-server/conf/s3.yaml`。可参考如下：
```
Comment (Radeity) [inline docs/docs/zh/guide/resource/configuration.md]: Ditto.

**Review by Gallardot — CHANGES_REQUESTED** [review]
Rest LGTM

File: docs/docs/en/guide/resource/configuration.md (hunk 2 above)
Comment (ruanwenjun) [reply docs/docs/en/guide/resource/configuration.md]: Done

File: docs/docs/zh/guide/resource/configuration.md (hunk 3 above)
Comment (ruanwenjun) [reply docs/docs/zh/guide/resource/configuration.md]: Done

File: deploy/kubernetes/dolphinscheduler/values.yaml (hunk 1 above)
Comment (ruanwenjun) [reply deploy/kubernetes/dolphinscheduler/values.yaml]: Done, add `aws`

**Review by Radeity — APPROVED** [review]
LGTM

**Review by Gallardot — APPROVED** [review]
LGTM

**Review by rickchengx — APPROVED** [review]
LGTM

**Pull request merged**